WordPress/gutenberg
# Global style revisions: move change summary code and tests to block editor package

## What?

- Moves `getRevisionChanges` and tests from the edit site package to the block editor packages.
- Renames the method to `getGlobalStylesChangelist`

### TODO

- [x] Move [max-length functionality from the revision buttons](https://github.com/WordPress/gutenberg/blob/cd912140868df0c652ccc04c2b4eb52d4ec23410/packages/edit-site/src/components/global-styles/screen-revisions/revisions-buttons.js#L40-L40) to the util.
- [x] Refactor to memoize `getBlockTypes` transformation
- [x] Add tests for the above

Related to:

- https://github.com/WordPress/gutenberg/issues/55776

## Why?
So that the editor can use this utility elsewhere, namely, in the save entity panel to describe global styles changes in the sidebar.



## Testing Instructions

Check that there are no regressions in global styles revisions changes summaries.

The original PR was https://github.com/WordPress/gutenberg/pull/56577

Testing instructions from that PR:

> Make a bunch of changes to your global styles revisions, ensuring to get a decent mix of block, top-level, element styles and layout, typography and color settings.
> 
> If the theme, e.g., 2024, has style variations, activate and save a few revision using those.
> 
> Open up the styles revisions panel and check that the change summary makes sense.
